src/pages/Swap/index.tsx

```diff
@@ -463,6 +463,33 @@ export function Swap({
     tradeToConfirm,
   ])
 
+  const handleOnWrap = useCallback(() => {
+    if (!onWrap) return
+    onWrap()
+      .then((hash) => {
+        setSwapState((currentState) => ({
+          ...currentState,
+          swapError: undefined,
+          txHash: hash,
+        }))
+        onUserInput(Field.INPUT, '')
+      })
+      .catch((error) => {
+        if (!didUserReject(error)) {
+          sendAnalyticsEvent(SwapEventName.SWAP_ERROR, {
+            wrapType,
+            input: currencies[Field.INPUT],
+            output: currencies[Field.OUTPUT],
+          })
+        }
+        setSwapState((currentState) => ({
+          ...currentState,
+          swapError: error,
+          txHash: undefined,
+        }))
+      })
```

Comment:
I'd prefer this use async-await syntax for readability/maintainability:

```suggestion
    if (!onWrap) return
    try {
      const txHash = await onWrap()
      setSwapState((currentState) => ({
        ...currentState,
        swapError: undefined,
        txHash,
      }))
      onUserInput(Field.INPUT, '')
    } catch (error) {
      if (!didUserReject(error)) {
        sendAnalyticsEvent(SwapEventName.SWAP_ERROR, {
          wrapType,
          input: currencies[Field.INPUT],
          output: currencies[Field.OUTPUT],
        })
      }
      setSwapState((currentState) => ({
        ...currentState,
        swapError: error,
        txHash: undefined,
      }))
    }
```